templates/intel_pmu.conf.j2

```diff
@@ -10,8 +10,9 @@ LoadPlugin intel_pmu
 {% if collectd_plugin_intelpmu_cores is defined %}
   Cores{% for item in collectd_plugin_intelpmu_cores %} "{{ item }}"{% endfor %}
 {% endif %}
-
+{% if collectd_plugin_intelpmu_reporthardwarecacheevents is defined %}
   ReportHardwareCacheEvents {{ collectd_plugin_intelpmu_reporthardwarecacheevents | string | lower }}
+{% endif %}
   ReportKernelPMUEvents {{ collectd_plugin_intelpmu_reportkernelpmuevents | string | lower }}
   ReportSoftwareEvents {{ collectd_plugin_intelpmu_reportsoftwareevents | string | lower }}
 {% if collectd_plugin_intelpmu_eventlist is defined %}
```

Comment:
Couple comments about the below section (I know not part of this review...)

The event_list seems like it should be a list of all the events in a single argument surrounded by double quotes?

* https://github.com/voxpupuli/puppet-collectd/blob/master/templates/plugin/intel_pmu.conf.erb#L11-L13

Also the HardwareEvents looks like it should be a single argument in double quotes, with all list items joined by a comma

* https://github.com/voxpupuli/puppet-collectd/blob/master/templates/plugin/intel_pmu.conf.erb#L14-L16

Reply:
I'll have to take a close look at that, I've added a new issue for it [1] and will probably pull this template into a separate branch/PR.

[1] https://github.com/infrawatch/collectd-config-ansible-role/issues/29